scene_synchronizer.cpp

```diff
@@ -2387,20 +2387,13 @@ void ServerSynchronizer::generate_snapshot_node_data(
 		snap_node_data = p_node_data->id;
 	}
 
-	if ((node_has_changes && skip_snapshot_variables == false) || force_snapshot_node_path || include_input_id || unknown) {
+	if ((node_has_changes && skip_snapshot_variables == false) || force_snapshot_node_path || unknown) {
 		r_snapshot_data.push_back(snap_node_data);
 	} else {
 		// It has no changes, skip this node.
 		return;
 	}
 
-	if (include_input_id) {
-		NetworkedController *controller = static_cast<NetworkedController *>(p_node_data->node);
-
-		// This is a controller, always sync it.
-		r_snapshot_data.push_back(controller->get_current_input_id());
```

Comment:
Yup. It can happen in two cases:
1. The controller exists on the client and the received snapshot doesn't contains an InputID: **An error is raised and the snapshot is considered corrupted**.
2. The controller DOESN'T exists on the client and the received snapshot doesn't contains an InputID: **The snapshot is right away applied.**